XCUITests/DomainAutocompleteTest.swift

```diff
@@ -0,0 +1,175 @@
+        navigator.goto(URLBarOpen)
+        app.textFields["address"].typeText("w")
+
+        waitForValueContains(app.textFields["address"], value: website["value"]!)
+        let value = app.textFields["address"].value
+        XCTAssertEqual(value as? String, website["value"]!, "Wrong autocompletion")
+
+        // Enter the complete website and check that there is not more text added, just what user typed
+        app.buttons["Clear text"].tap()
+        app.textFields["address"].typeText(website["value"]!)
+        waitForValueContains(app.textFields["address"], value: website["value"]!)
+        let value2 = app.textFields["address"].value
+        XCTAssertEqual(value2 as? String, website["value"]!, "Wrong autocompletion")
+    }
+    // Test that deleting characters works correctly with autocomplete
+    func testAutocompleteDeletingChars() {
+        navigator.openURL(urlString: website["url"]!)
+        navigator.goto(URLBarOpen)
+        app.textFields["address"].typeText("www.moz")
+
+        // First delete the autocompleted part
+        app.textFields["address"].typeText("\u{0008}")
+        // Then remove an extra char and check that the autocompletion stops working
+        app.textFields["address"].typeText("\u{0008}")
+        waitForValueContains(app.textFields["address"], value: "mo")
+        // Then write another letter and the autocompletion works again
+        app.textFields["address"].typeText("z")
+        waitForValueContains(app.textFields["address"], value: "moz")
+
+        let value = app.textFields["address"].value
+        XCTAssertEqual(value as? String, website["value"]!, "Wrong autocompletion")
+    }
+    // Delete the entire string and verify that the home panels are shown again.
+    func testDeleteEntireString() {
+        navigator.openURL(urlString: website["url"]!)
+        navigator.goto(URLBarOpen)
+        app.textFields["address"].typeText("www.moz")
+        waitforExistence(app.buttons["Clear text"])
+        app.buttons["Clear text"].tap()
+
+        // Check that the address field is empty and that the home panels are shown
+        let value = app.textFields["address"].value
+        XCTAssertEqual(value as? String, "", "The url has not been removed correctly")
+
+        waitforExistence(app.buttons["HomePanels.TopSites"])
+        XCTAssertFalse(app.buttons["HomePanels.TopSites"].isEnabled)
+        XCTAssertTrue(app.buttons["HomePanels.Bookmarks"].isEnabled)
+    }
+    // Ensure that the scheme is included in the autocompletion.
+    func testEnsureSchemeIncludedAutocompletion() {
+        navigator.openURL(urlString: website["url"]!)
+        navigator.goto(URLBarOpen)
+        app.textFields["address"].typeText("https")
+        waitForValueContains(app.textFields["address"], value: "mozilla")
+        let value = app.textFields["address"].value
+        XCTAssertEqual(value as? String, "https://www.mozilla.org/", "Wrong autocompletion")
+    }
+    // Multiple subdomains. TBDone
```

Comment:
Hmm, they injected the mutiple subdomain example in the UITests, then, how about we enable the UITest with only this one enabled? so the test will be like:
```

tester().tapView(withAccessibilityIdentifier: "url")
let textField = tester().waitForView(withAccessibilityLabel: "Address and Search") as! UITextField

// Ensure that the scheme is included in the autocompletion.
        tester().enterText(intoCurrentFirstResponder: "https")
        BrowserUtils.ensureAutocompletionResult(tester(), textField: textField, prefix: "https", completion: "://foo.bar.baz.org/")
        tester().clearTextFromFirstResponder()

         // Multiple subdomains.
         tester().enterText(intoCurrentFirstResponder: "f")
         BrowserUtils.ensureAutocompletionResult(tester(), textField: textField, prefix: "f", completion: "oo.bar.baz.org/")
         BrowserUtils.ensureAutocompletionResult(tester(), textField: textField, prefix: "ba", completion: "r.baz.org/")
         tester().enterText(intoCurrentFirstResponder: "z")
         BrowserUtils.ensureAutocompletionResult(tester(), textField: textField, prefix: "baz", completion: ".org/")
```